networks/p2p/server.go

```diff
@@ -577,8 +577,9 @@ func (srv *MultiChannelServer) setupConn(c *conn, flags connFlag, dialDest *disc
 			dialDest.TCPs = append(dialDest.TCPs, uint16(listenPort))
 		}
 		srv.AddPeer(dialDest)
-		logger.Info("[Dial] Try to update dial candidate with a multichannel peer", "id", dialDest.ID, "addr", dialDest.IP, "port", dialDest.TCPs)
-		return nil
+		logger.Info("[Dial] update and retry the dial candidate as a multichannel",
+			"id", dialDest.ID, "addr", dialDest.IP, "port", dialDest.TCPs)
+		return errUpdateDial
```

Comment:
So, adding this error closes `fd`. Is this on purpose?

Reply:
When an update occurs, `dial` is called instead of `dialMulti` and `fd` is not closed. However, I think a fd should be closed when `dial` fails, too. I added TODO for this case.

Reply:
Thanks :)

Reply:
Sorry for confusing answer, above. 
Both `MultiChannelServer` and `BaseServer` calls `c.close()` and they both add `DiscReason`.